x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/attribute_selector.tsx

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import {
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiPanel,
+  EuiSelect,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+
+import {
+  ANY_AUTH_PROVIDER,
+  ANY_AUTH_PROVIDER_OPTION_LABEL,
+  AUTH_ANY_PROVIDER_LABEL,
+  AUTH_INDIVIDUAL_PROVIDER_LABEL,
+  ATTRIBUTE_SELECTOR_TITLE,
+  AUTH_PROVIDER_LABEL,
+  EXTERNAL_ATTRIBUTE_LABEL,
+  ATTRIBUTE_VALUE_LABEL,
+} from './constants';
+
+interface IAttributeSelectorProps {
+  attributeName: string;
+  attributeValue?: string;
+  attributes: string[];
+  selectedAuthProviders?: string[];
+  availableAuthProviders?: string[];
+  elasticsearchRoles: string[];
+  disabled: boolean;
+  multipleAuthProvidersConfig: boolean;
+  handleAttributeSelectorChange(value: string, elasticsearchRole: string): void;
+  handleAttributeValueChange(value: string): void;
+  handleAuthProviderChange?(value: string[]): void;
+}
+
+interface AttributeExamples {
+  username: string;
+  email: string;
+  metadata: string;
+}
+
+interface ParentOption extends EuiComboBoxOptionOption<string> {
+  label: string;
+  options: ChildOption[];
+}
+
+interface ChildOption extends EuiComboBoxOptionOption<string> {
+  value: string;
+  label: string;
+}
+
+const attributeValueExamples = {
+  username: 'elastic,*_system',
+  email: 'user@example.com,*@example.org',
+  metadata: '{"_reserved": true}',
+};
+
+const getAuthProviderOptions = (availableAuthProviders: string[]) => {
+  return [
+    {
+      label: AUTH_ANY_PROVIDER_LABEL,
+      options: [{ value: ANY_AUTH_PROVIDER, label: ANY_AUTH_PROVIDER_OPTION_LABEL }],
+    },
+    {
+      label: AUTH_INDIVIDUAL_PROVIDER_LABEL,
+      options: availableAuthProviders.map((authProvider) => ({
+        value: authProvider,
+        label: authProvider,
+      })),
+    },
+  ];
+};
+
+const getSelectedOptions = (selectedAuthProviders: string[], availableAuthProviders: string[]) => {
```

Comment:
When I first saw this, I thought it may be a good candidate for `useMemo` ... but since it should be a relatively small data set it's probably not worth it. Just thinking out loud.